### PR TITLE
Use an Informer to list LimitRanges 🥼

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

Using a LimitRange lister here instead, so this doesn't end up hitting
the real API server on each call.

Taking into account a review :
https://github.com/tektoncd/pipeline/pull/4176#issuecomment-920366660.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/kind cleanup
/cc @imjasonh @bobcatfish @pritidesai @sbwsg @afrittoli 

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by vdemeester** [comment]
I am missing something very obvious in the unit tests.. but.. for some reason I don't see it..  😅

File: pkg/internal/limitrange/transformer_test.go
Comment (imjasonh) [inline pkg/internal/limitrange/transformer_test.go:29]: Can remove this

**Review by imjasonh — COMMENTED** [review]
/approve

File: internal/builder/v1beta1/pod.go
Comment (vdemeester) [inline internal/builder/v1beta1/pod.go:129]: I am not sure why this was needed to be changed to fix the issue with some tests but... meh.

File: pkg/reconciler/taskrun/taskrun_test.go
Comment (vdemeester) [inline pkg/reconciler/taskrun/taskrun_test.go:1817]: A general rule of thumbs in theses tests : *never use `context.Background`* as `context.Context` for the `Reconcile` loop, as it bypass **all* the init magic `knative/pkg` does 😅

**Review by sbwsg — APPROVED** [review]

**Review by afrittoli — COMMENTED** [review]
Thanks for this!
/lgtm

File: pkg/reconciler/taskrun/taskrun_test.go
Comment (mattmoor) [reply pkg/reconciler/taskrun/taskrun_test.go:1817]: So there is a little bit of magic this does, but the clients and informers are only injected on the `ctx` passed to `NewController` methods.  The `.Get()` methods aren't free (they are a list walk vs. field lookup), so we want folks to stash these during initialization and access them via fields.

So I'd say `context.Background()` is actually probably fine here, just not in calls to `NewController` :sweat`

File: pkg/internal/limitrange/transformer_test.go
Comment (mattmoor) [inline pkg/internal/limitrange/transformer_test.go:1015]: Could consumers just use `fakelimitrangeinformer.Get(ctx)` instead of returning it here?

**Review by mattmoor — COMMENTED** [review]
/lgtm

This LGTM, just a nit about `setup` if you want to preserve the signature.

**Comment by mattmoor** [comment]
/hold

In case you want to do the nit, otherwise feel free to remove

File: pkg/internal/limitrange/transformer_test.go
Comment (vdemeester) [reply pkg/internal/limitrange/transformer_test.go:1015]: c'est pas faux 😝

**Review by mattmoor — COMMENTED** [review]
/lgtm
/hold cancel

File: pkg/reconciler/taskrun/taskrun_test.go
Comment (vdemeester) [reply pkg/reconciler/taskrun/taskrun_test.go:1817]: @mattmoor yeah, it's a bit tricky. The whole file would benefit from a small refactoring.

File: pkg/internal/limitrange/transformer_test.go
Comment (vdemeester) [reply pkg/internal/limitrange/transformer_test.go]: > c'est pas faux stuck_out_tongue_closed_eyes

Man I am tired.. "c'est pas faux" means "it's not wrong".. speaking in french while commenting.. makes this.. 😅

**Pull request merged**
